tests/ci_tests/code_formatter/crewai/test_crewai_agents.py

```diff
@@ -0,0 +1,310 @@
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        result = agents.create_code_formatter_agents(mock_langfuse_client)
+
+        assert isinstance(result, dict)
+        assert "senior_engineer" in result
+        assert isinstance(result["senior_engineer"], Agent)
+
+    def test_agent_has_correct_role(self):
+        """Test that the senior engineer agent has correct role"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal text"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory text"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        assert agent.role == "Senior Software Engineer"
+
+    def test_agent_uses_langfuse_prompts(self):
+        """Test that agent uses prompts from Langfuse"""
+        mock_langfuse_client = mock.MagicMock()
+
+        expected_goal = "Refactor code according to best practices"
+        expected_backstory = "You have extensive experience in Python development"
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = expected_goal
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = expected_backstory
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
+        # Verify the prompts were fetched
+        assert mock_langfuse_client.get_prompt.call_count == 2
+        mock_langfuse_client.get_prompt.assert_any_call("refactor_agent/goal_prompt")
+        mock_langfuse_client.get_prompt.assert_any_call("refactor_agent/backstory_prompt")
+
+        # Verify the agent uses the prompts
+        assert agent.goal == expected_goal
+        assert agent.backstory == expected_backstory
+
+    def test_agent_has_no_tools(self):
+        """Test that the senior engineer agent has no tools"""
+        mock_langfuse_client = mock.MagicMock()
+
+        mock_goal_prompt = mock.MagicMock()
+        mock_goal_prompt.prompt = "goal"
+        mock_backstory_prompt = mock.MagicMock()
+        mock_backstory_prompt.prompt = "backstory"
+
+        mock_langfuse_client.get_prompt.side_effect = [
+            mock_goal_prompt,
+            mock_backstory_prompt,
+            "template",
+            "template",
+        ]
+
+        agent_map = agents.create_code_formatter_agents(mock_langfuse_client)
+        agent = agent_map["senior_engineer"]
+
```

Comment:
Added a docstring to the method, describing its purpose.
```suggestion
    def test_agent_does_not_delegate(self):
```